From 21eb78f2f655b34a6b55f0d3d76f2523732f2666 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Tue, 5 Nov 2013 22:25:59 +0000
Subject: [PATCH] builder: Fail if bison is not installed.

This partially fixes commit a4800e2d4fc50f372a8b626333c4fbb9b8ca9633.

(cherry picked from commit f526fe1766471b51b92cad74af0103d1889db83a)
---
 configure.ac | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 56a7eb5..f6a08c9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -711,9 +711,14 @@ PKG_CHECK_MODULES([LIBLZMA], [liblzma],[
 ],
 [AC_MSG_WARN([liblzma not found, virt-builder will be slower])])
 
-dnl (f)lex and bison are required for virt-builder.
+dnl (f)lex and bison for virt-builder (required).
+dnl XXX Could be optional with some work.
 AC_PROG_LEX
 AC_PROG_YACC
+dnl These macros don't fail, instead they set some useless defaults.
+if test "x$YACC" = "xyacc"; then
+    AC_MSG_FAILURE([GNU 'bison' is required (yacc won't work).])
+fi
 
 dnl Check for QEMU for running binaries on this $host_cpu, fall
 dnl back to basic 'qemu'.  Allow the user to override it.
-- 
1.8.4.2

